### ISSUE TITLE
poor accuracy of cpu / ocl remap

### ISSUE THREAD

**Issue by Ginkgo-Biloba** [opencv/opencv#26361]
### System Information

OpenCV version: 4.8.0.

But the 4.x branch will gets same results.

### Detailed description

`cv::remap` (cpu / ocl) uses look-up table for interpolation, and INTER_BITS is 5, i.e. the table size is 32. Thus the positions' (weights') precision of bi-linear is `0.5/32+0.499999999/32 → 1/32 = 3.125% ≈ 8/255`.

`cv::cuda::remap` uses raw interpolation, so its' result is well.

This problem also exists in warpAffine and warpPerspective.

https://github.com/opencv/opencv/blob/331412dfadc3c270317854ccc4c53f17cc35e15f/modules/imgproc/include/opencv2/imgproc.hpp#L290C1-L296C1

### Steps to reproduce

data
[reamp.tgz](https://github.com/user-attachments/files/17495728/reamp.tgz)

Code to dump result. Just a simple resize with border.

``` cpp
#define _CRT_SECURE_NO_WARNINGS
#include <opencv2/imgproc.hpp>
#include <opencv2/cudawarping.hpp>

#pragma comment(lib, "opencv_cudawarping4100d")
#pragma comment(lib, "opencv_imgproc4100d")
#pragma comment(lib, "opencv_cudev4100d")
#pragma comment(lib, "opencv_core4100d")

using namespace cv;
static float const affine[4] = {2.778427, -18.006325, 2.789243, 110.456535};

static void dumpbin(Mat const& a, String const& name)
{
	CV_Assert(a.isContinuous());
	FILE* fid = fopen(name.data(), "wb");
	fwrite(a.data, a.elemSize(), a.total(), fid);
	fclose(fid);
}

int main()
{
	Mat src, dst, x, y;
	UMat usrc, udst, ux, uy;
	cuda::GpuMat gsrc, gdst, gx, gy;

	// src
	Mat image(1080, 1920, CV_8UC3);
	FILE* fid = fopen("image.rgb24", "rb");
	fread(image.data, image.elemSize(), image.total(), fid);
	fclose(fid);
	image.convertTo(src, CV_32F);

	// grid
	x.create(352, 704, CV_32F);
	y.create(352, 704, CV_32F);
	AutoBuffer<float> X(x.cols), Y(x.rows);
	for (int i = 0; i < x.cols; ++i)
		X[i] = i * affine[0] + affine[1];
	for (int i = 0; i < x.rows; ++i)
		Y[i] = i * affine[2] + affine[3];
	for (int h = 0; h < x.rows; ++h) {
		float* xptr = x.ptr<float>(h);
		float* yptr = y.ptr<float>(h);
		for (int w = 0; w < x.cols; ++w)
			xptr[w] = X[w], yptr[w] = Y[h];
	}

	// cpu
	cv::remap(src, dst, x, y, INTER_LINEAR, BORDER_CONSTANT);
	dumpbin(dst, "cvcpu");
	// cuda
	gsrc.upload(src);
	gx.upload(x);
	gy.upload(y);
	cv::cuda::remap(gsrc, gdst, gx, gy, INTER_LINEAR, BORDER_CONSTANT);
	gdst.download(dst);
	dumpbin(dst, "cvcuda");
	// opencl
	src.copyTo(usrc);
	x.copyTo(ux);
	y.copyTo(uy);
	cv::remap(usrc, udst, ux, uy, INTER_LINEAR, BORDER_CONSTANT);
	udst.copyTo(dst);
	dumpbin(dst, "cvocl");
}

```

code to compare

``` python
import numpy as np
import torch
from matplotlib import pyplot
from PIL import Image


src: np.ndarray
dst: np.ndarray
X: np.ndarray
Y: np.ndarray

affine = np.array([2.778427, -18.006325, 2.789243, 110.456535], np.float32)
src = np.fromfile("image.rgb24", np.uint8)
src = src.astype(np.float32).reshape(1080, 1920, 3)
X = np.arange(704, dtype=np.float32)
Y = np.arange(352, dtype=np.float32)
# warp affine by torch
X = 2 * (X * affine[0] + affine[1] + 0.5) / src.shape[1] - 1
Y = 2 * (Y * affine[2] + affine[3] + 0.5) / src.shape[0] - 1
# np.float32
print(src.dtype, X.dtype)
tsrc = torch.tensor(np.transpose(src, (2, 0, 1)))
grid = torch.tensor(np.stack(np.meshgrid(X, Y), axis=2))
tsrc = tsrc.view(1, *tsrc.shape)
grid = grid.view(1, *grid.shape)
tdst = torch.nn.functional.grid_sample(
	tsrc, grid, mode="bilinear", padding_mode="zeros", align_corners=False
)
dst = np.ascontiguousarray(tdst[0].permute(1, 2, 0).numpy())
dst.tofile("torch")

data = [
	("torch", dst),
	("cvcpu", np.fromfile("cvcpu", np.float32).reshape(dst.shape)),
	("cvocl", np.fromfile("cvocl", np.float32).reshape(dst.shape)),
	("cvcuda", np.fromfile("cvcuda", np.float32).reshape(dst.shape)),
]
# ignore the border
droi = slice(20, -20)
for i in range(len(data)):
	for k in range(i + 1, len(data)):
		d = np.abs(data[i][1] - data[k][1])
		e = d[droi, droi]
		print(f"{data[i][0]:6} vs {data[k][0]:6}  {d.max()}  {e.max()}")

d = np.clip(np.abs(data[0][1] - data[1][1]) * 60, None, 255).astype(np.uint8)
Image.fromarray(d).save("torch-cvcpu.jpg")
pyplot.imshow(d)
```

outputs

```
torch  vs cvcpu   4.541175842285156  4.541175842285156
torch  vs cvocl   4.541175842285156  4.541175842285156
torch  vs cvcuda  0.014190673828125  0.014190673828125
cvcpu  vs cvocl   0.0  0.0
cvcpu  vs cvcuda  4.538719177246094  4.538719177246094
cvocl  vs cvcuda  4.538719177246094  4.538719177246094
```

The difference is more obvious on image edges.
![torch-cvcpu](https://github.com/user-attachments/assets/6d01443c-6918-45c8-8fb2-3f9216cb0e3b)

### Issue submission checklist

- [X] I report the issue, it's not a question
- [X] I checked the problem with documentation, FAQ, open issues, forum.opencv.org, Stack Overflow, etc and have not found any solution
- [X] I updated to the latest OpenCV version and the issue is still there
- [X] There is reproducer code and related data files (videos, images, onnx, etc)

**Comment by YanBC** [comment]
Duplicate of #25937. A fix is on its way.

**Comment by asmorkalov** [comment]
Addressed in https://github.com/opencv/opencv/pull/26348
